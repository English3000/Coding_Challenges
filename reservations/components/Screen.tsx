import React from "react"
import { AppRegistry, ScrollView, View, Text } from "react-native"

type Errors = {
  graphQLErrors : [{message : string}]
}
type Props = {
  component : () => JSX.Element,
  error    ?: Errors,
  loading   : boolean,
}
export default function Screen({loading, error, component} : Props){
  return (
    <View>
      {loading ?
        <View><Text>Loading...</Text></View>
      : null}

      {error ?
        <ScrollView>
          {error.graphQLErrors.map(({message}, index) =>
            <Text key={index}>{message}</Text>
          )}
        </ScrollView>
      : null}

      {!loading && !error ? component() : null}
    </View>
  )
}
AppRegistry.registerComponent("Screen", () => Screen)