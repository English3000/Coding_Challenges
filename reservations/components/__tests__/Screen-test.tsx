import React from "react"
import { View } from "react-native"
import Screen from "../Screen"
import { create } from "react-test-renderer"

it("loads", () => {
  expect(
    create(<Screen loading={true} component={() => <View></View>}/>)
  ).toMatchSnapshot()
})

it("handles error", () => {
  const error = { graphQLErrors: [{message: "not working"}] }
  // @ts-ignore
  create(<Screen loading={false} error={error} component={() => <View></View>}/>)
    .root
    .findByProps({error})
})

it("renders", () => {
  // @ts-ignore (`id` used strictly for testing)
  create(<Screen loading={false} component={() => <View id="test"></View>}/>)
    .root
    .findByProps({id: "test"})
})