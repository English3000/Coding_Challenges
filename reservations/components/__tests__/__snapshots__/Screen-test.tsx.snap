// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loads 1`] = `
<View>
  <View>
    <Text>
      Loading...
    </Text>
  </View>
</View>
`;

exports[`loads correctly 1`] = `
<View>
  <View>
    <Text>
      Loading...
    </Text>
  </View>
</View>
`;

exports[`renders correctly 1`] = `
<View>
  <View>
    <Text>
      Loading...
    </Text>
  </View>
</View>
`;
